### PR TITLE
No issue: fixes test broken by snapshot

### PR DESCRIPTION
**Pull request by Baron-Severin**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/components/StoreProviderTest.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ class StoreProviderTest {
 
     private class BasicState : State
 
-    private val basicStore = Store(BasicState()) { state, _: Action -> state }
+    private val basicStore = Store(BasicState(), { state, _: Action -> state })
```
Comment (pocmo) [inline app/src/test/java/org/mozilla/fenix/components/StoreProviderTest.kt]: Oh, sorry, I added a parameter to the store constructor and didn't expect that to cause any issues. Didn't think of _trailing lambda_ usage.

Comment (Baron-Severin) [reply app/src/test/java/org/mozilla/fenix/components/StoreProviderTest.kt]: All good @pocmo, it's a weird edge case.
